### PR TITLE
Refactor auth code in modules

### PR DESCRIPTION
**Pull request by rhysmeister**
##### SUMMARY
Implement https://github.com/ansible-collections/community.mongodb/issues/377 in appropriate modules.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
All modules

### CODE REVIEW

File: plugins/modules/mongodb_oplog.py [hunk 1]
```diff
@@ -182,68 +183,51 @@ def main():
...
-                module.fail_json(msg='Unable to get MongoDB server version: %s' % to_native(excep))
-
-            # Get driver version::
-            driver_version = LooseVersion(PyMongoVersion)
-            # Check driver and server version compatibility:
-            check_compatibility(module, srv_version, driver_version)
-        except Exception as excep:
-            module.fail_json(msg='Unable to authenticate with MongoDB: %s' % to_native(excep))
+    try:
+        srv_version = LooseVersion(client.server_info()['version'])
```
Comment (cognifloyd) [inline plugins/modules/mongodb_oplog.py:189]: Can you use mongodb_common.check_srv_version here to get the version?

File: plugins/modules/mongodb_replicaset.py [hunk 2]
```diff
@@ -433,16 +433,15 @@ def main():
...
     try:
-        rs = replicaset_find(client)
+        rs = replicaset_find(client)  # does not require auth
     except Exception as e:
         module.fail_json(msg='Unable to connect to query replicaset: %s' % to_native(e))
 
     if isinstance(rs, str):
         if replica_set == rs:
             if reconfigure:
+                mongo_auth(module, client)
```
Comment (cognifloyd) [inline plugins/modules/mongodb_replicaset.py:444]: I think you also need `mongo_auth` just before the `replicaset_add` call below.

**Review by cognifloyd — COMMENTED** [review]
What about your module template? I think that meds to be adjusted too. https://github.com/ansible-collections/community.mongodb/blob/master/docs/mongodb_module.template.py

Plus a few other items I noticed while reading through these changes.

File: plugins/modules/mongodb_replicaset.py (hunk 2 above)
Comment (rhysmeister) [reply plugins/modules/mongodb_replicaset.py:444]: Can you actually enable auth before initiating a replicaset? It's been such a long time I don't really remember.

File: plugins/modules/mongodb_oplog.py (hunk 1 above)
Comment (rhysmeister) [reply plugins/modules/mongodb_oplog.py:189]: Yes, I actually meant to have done that.

File: plugins/modules/mongodb_replicaset.py (hunk 2 above)
Comment (rhysmeister) [reply plugins/modules/mongodb_replicaset.py:444]: Yes... https://docs.mongodb.com/manual/tutorial/deploy-replica-set-with-keyfile-access-control/

Comment (rhysmeister) [reply plugins/modules/mongodb_replicaset.py:444]: I think this is correct. Resolving.

**Comment by rhysmeister** [comment]
Just wanted to bump that coverage % back up. Merging!

**Pull request merged**